src/simulator/services/persistent_state.py

```diff
@@ -0,0 +1,184 @@
+
+from structures import DynamicColour, Colour, TRANSPARENT
+from simulator.services.service import Service
+from simulator.services.services import Services
+from simulator.services.event_manager.events.colour_update_event import ColourUpdateEvent
+from simulator.services.event_manager.events.new_colour_event import NewColourEvent
+from simulator.services.debug import DebugLevel
+from utility.utils import exclude_from_dict
+
+from typing import Dict, Final, Any, List, Optional
```

Comment:
`Final` doesn't work (ImportError) on Python <3.8, can we use the fix in `src/structures.py` for now and talk about a better fix later?

Reply:
Fixed.